File: java/InterstitialExample/app/src/main/java/com/pubscale/sdkone/example/interstitialexample/MainActivity.java
package com.pubscale.sdkone.example.interstitialexample;

import android.os.Bundle;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.pubscale.sdkone.core.app_open_ads.general.GGAppOpenAds;
import com.pubscale.sdkone.core.interstitial.general.GGInterstitialAd;
import com.pubscale.sdkone.core.interstitial.general.GGInterstitialEventsListener;
import com.pubscale.sdkone.core.models.general.AdErrors;
import com.pubscale.sdkone.example.interstitialexample.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        //Disabling auto app open ads
        GGAppOpenAds.setShouldShowOnAppMovedToForeground(false);

        binding.btnLoad.setOnClickListener(v -> {
            loadInterstitialAd();
        });
        binding.btnShow.setOnClickListener(v -> {
            showAd();
        });
    }

    private GGInterstitialAd mAd = null;

    private void loadInterstitialAd() {
        mAd = new GGInterstitialAd(this, "float-13569");

        binding.tvStatus.setText("Ad Loading");

        mAd.addListener(new GGInterstitialEventsListener() {
            @Override
            public void onAdLoaded() {
                binding.tvStatus.setText("Ad Loaded");
                binding.btnShow.setEnabled(true);
            }

            @Override
            public void onAdLoadFailed(@NonNull AdErrors adErrors) {
                mAd.removeListener(this);
                binding.tvStatus.setText("Ad Load Failed - " + adErrors.name());
            }

            @Override
            public void onAdShowFailed() {
                mAd.removeListener(this);
                binding.tvStatus.setText("Ad Show Failed");
            }

            @Override
            public void onAdOpened() {
                Toast.makeText(MainActivity.this, "Ad Opened", Toast.LENGTH_SHORT).show();
                binding.tvStatus.setText("Ad Opened");
            }

            @Override
            public void onAdClosed() {
                binding.tvStatus.setText("Ad Closed");
                binding.btnShow.setEnabled(false);

                //Remove listener to prevent memory leaks
                mAd.removeListener(this);
            }
        });

        //Load ad after adding a listener
        mAd.loadAd();
    }

    private void showAd() {
        //Show the ad
        mAd.show(this);
    }
}

File: kotlin/InterstitialExample/app/src/main/java/com/pubscale/sdkone/example/interstitialexample/MainActivity.kt
package com.pubscale.sdkone.example.interstitialexample

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.pubscale.sdkone.core.app_open_ads.general.GGAppOpenAds
import com.pubscale.sdkone.core.interstitial.general.GGInterstitialAd
import com.pubscale.sdkone.core.interstitial.general.GGInterstitialEventsListener
import com.pubscale.sdkone.core.models.general.AdErrors
import com.pubscale.sdkone.example.interstitialexample.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        //Disabling auto app open ads
        GGAppOpenAds.setShouldShowOnAppMovedToForeground(false)

        with(binding) {
            btnLoad.setOnClickListener { loadInterstitialAd() }
            btnShow.setOnClickListener { showAd() }
        }
    }

    private val mAd = GGInterstitialAd(this, "float-13569")

    private fun loadInterstitialAd() {
        with(binding){
            tvStatus.text = "Ad Loading"

            mAd.addListener(object : GGInterstitialEventsListener {
                override fun onAdLoaded() {
                    tvStatus.text = "Ad Loaded"
                    btnShow.isEnabled = true
                }

                override fun onAdLoadFailed(adErrors: AdErrors) {
                    tvStatus.text = "Ad Load Failed - " + adErrors.name

                    //Remove listener to prevent memory leaks
                    mAd.removeListener(this)
                }

                override fun onAdShowFailed() {
                    tvStatus.text = "Ad Show Failed"

                    //Remove listener to prevent memory leaks
                    mAd.removeListener(this)
                }

                override fun onAdOpened() {
                    Toast.makeText(this@MainActivity, "Ad Opened", Toast.LENGTH_SHORT).show()
                    tvStatus.text = "Ad Opened"
                }

                override fun onAdClosed() {
                    tvStatus.text = "Ad Closed"
                    btnShow.isEnabled = false

                    //Remove listener to prevent memory leaks
                    mAd.removeListener(this)
                }
            })
        }

        //Load ad after adding a listener
        mAd.loadAd()
    }

    private fun showAd() {
        mAd.show(this)
    }
}